### ISSUE TITLE
use dump/raw or dump/norm

### ISSUE THREAD

**Issue by Mingrg** [kan-bayashi/ParallelWaveGAN#227]
Hi，
   As we all know that there is raw and norm in dump dir, What's the difference between **use dump/raw to train** and **use dump/norm to train**. I also tried to use dump/raw to train my own model, and it worked not bad. So I have two questions:
   1. In your code, I see that you use the dump/norm to train, did norm have a better performance in training pwg than raw? or is there any other reason for that?
   2. If I use your pre-trained model in resume to train my own dataset, should I use norm rather than raw?

**Comment by kan-bayashi** [comment]
`raw`: the raw mel-spectrogram
`norm`: the mel-spectrogram normalized with the statistics of the training data

> In your code, I see that you use the dump/norm to train, did norm have a better performance in training pwg than raw? or is there any other reason for that?

In my experience, using normalized features make it more robust to the input features.
Maybe if the input is clean, the performance is almost the same.
But if the input is mismatched with the natural features (e.g. voice conversion, TTS), the normalized model will give more stable output.

> If I use your pre-trained model in resume to train my own dataset, should I use norm rather than raw?

No. Please use norm calculated with the pretrained model's stats.
You can do it by `--pretrain` option.
https://github.com/kan-bayashi/ParallelWaveGAN/blob/53d14969089b3d3229fe6bfce221234f25a9d836/egs/template_single_spk/voc1/run.sh#L40

**Comment by Mingrg** [comment]
> `raw`: the raw mel-spectrogram
> `norm`: the mel-spectrogram normalized with the statistics of the training data
> 
> > In your code, I see that you use the dump/norm to train, did norm have a better performance in training pwg than raw? or is there any other reason for that?
> 
> In my experience, using normalized features make it more robust to the input features.
> Maybe if the input is clean, the performance is almost the same.
> But if the input is mismatched with the natural features (e.g. voice conversion, TTS), the normalized model will give more stable output.
> 
> > If I use your pre-trained model in resume to train my own dataset, should I use norm rather than raw?
> 
> No. Please use norm calculated with the pretrained model's stats.
> You can do it by `--pretrain` option.
> https://github.com/kan-bayashi/ParallelWaveGAN/blob/53d14969089b3d3229fe6bfce221234f25a9d836/egs/template_single_spk/voc1/run.sh#L40

Got it! Thank you for your detailed reply!

**Comment by Mingrg** [comment]
> `raw`: the raw mel-spectrogram
> `norm`: the mel-spectrogram normalized with the statistics of the training data
> 
> > In your code, I see that you use the dump/norm to train, did norm have a better performance in training pwg than raw? or is there any other reason for that?
> 
> In my experience, using normalized features make it more robust to the input features.
> Maybe if the input is clean, the performance is almost the same.
> But if the input is mismatched with the natural features (e.g. voice conversion, TTS), the normalized model will give more stable output.
> 
> > If I use your pre-trained model in resume to train my own dataset, should I use norm rather than raw?
> 
> No. Please use norm calculated with the pretrained model's stats.
> You can do it by `--pretrain` option.
> https://github.com/kan-bayashi/ParallelWaveGAN/blob/53d14969089b3d3229fe6bfce221234f25a9d836/egs/template_single_spk/voc1/run.sh#L40
> 
> If you specify it, the statistics calculation stage is skipped and copy the pretrained model's statistics.
> https://github.com/kan-bayashi/ParallelWaveGAN/blob/53d14969089b3d3229fe6bfce221234f25a9d836/egs/template_single_spk/voc1/run.sh#L106-L107

Sorry, one more question. What's the difference between "resume" and "pretrain"? I used to use "resume" to  resume training, but I never use "pretrain" before.

**Comment by kan-bayashi** [comment]
resume will load model, optimizer, and scheduler state but pretrain will load only model parameters.

**Comment by Mingrg** [comment]
> resume will load model, optimizer, and scheduler state but pretrain will load only model parameters.

Thanks!
